### ISSUE TITLE
Upgrade dependencies dry run - July 2021 

### ISSUE THREAD

**Issue by ghost** [ocaml/v3.ocaml.org-rescript#355]
The purpose of this task is to shake out any failures when upgrading dependencies and reduce how many release notes need to be reviewed in October 2021.

- [ ] Skim release notes since last global upgrade of dependencies, looking for any breaking changes impacting this site
- [ ] On a branch, upgrade all dependencies to the latest versions, except for any dependencies that we are being more cautious with
- [ ] On the same branch, apply any adjustments to code and configurations based on release note suggestions. Make note of each of these adjustments in the October 2021 upgrade issue.
- [ ] Perform manual regression test using the testing guide
- [ ] Create a follow-up task to finish this upgrade in October 2021

Skim release notes:
- [x] next-transpile-modules
- [x] next
- [x] tailwind/typography
- [x] autoprefixer
- [ ] tailwindcss

Apply a time box of 2 days to this task.

**Comment by ghost** [comment]
- [ ] Switch from bsc to new rescript cli wrapper as part of upgrading to >= 9.1.x
- [ ] Switch to using `.mjs` suffix and update webpack configuration to recognize `.mjs`, if this approach becomes the established standard for expressing that a project is implemented using ESM. (There is an example of doing this in an old PR (https://github.com/ocaml/v3.ocaml.org/pull/358/commits/78ff90c080b29f3e471495ebb9e5a6b855e16e42) as well as in the rescript-lang.org PR history around April 2021.)

**Comment by ghost** [comment]
- [ ] Upgrade to nodejs 15, if it is the new LTS version in July.

**Comment by ghost** [comment]
- [ ] Use the new JIT engine in tailwindcss.

This will be done in #385.

**Comment by ghost** [comment]
Let's implement a few unit tests for some of our bindings, so that we can upgrade dependency versions with fewer manual regression tests.

**Comment by ghost** [comment]
We may want to finish #298 before starting this issue, or we can postpone for a while longer until more of the universe including nextjs, has moved fully towards ESM.

**Comment by ghost** [comment]
If nextjs fully supports ESM by July, remove next-transpile-modules.

**Comment by ghost** [comment]
Here are some current warnings:
```
19:29:26.798 | warning " &gt; @glennsl/bs-json@5.0.2" has incorrect peer dependency "bs-platform@^5.0.4".
19:29:26.801 | warning " &gt; autoprefixer@10.2.5" has unmet peer dependency "postcss@^8.1.0".
19:29:26.801 | warning " &gt; tailwindcss@2.1.2" has unmet peer dependency "postcss@^8.0.9".
19:29:26.801 | warning "tailwindcss &gt; postcss-nested@5.0.5" has unmet peer dependency "postcss@^8.1.13".
```

The bs-json dependency has been fixed in the master branch of bs-json. It hasn't been published to npm yet.

**Comment by ghost** [comment]
- [ ] Upgrade esy to 0.6.10 explicitly

**Comment by ghost** [comment]
- [ ] Revert to using "dependencies" instead of "devDependencies" in package.json for most tools. (After reading more discussions online, I like the suggestion of having "dependencies" include all tools needed to build the site for npm packages that are building static sites, not just libraries that will be referenced at runtime in the packaged build.)

**Comment by ghost** [comment]
If we implemented backstop or Percyio, we could automatically merge upgrades to autoprefixer, tailwindcss, and tailwind/typography.

**Comment by ghost** [comment]
- [ ] Try out NextJS 11 "Conformance" validations - https://nextjs.org/blog/next-11#conformance

**Comment by ghost** [comment]
I think we are moving towards more frequent, ad-hoc upgrades of dependencies, so I will close this issue for now.
